src/sds_data_model/raster.py

```diff
@@ -55,11 +55,71 @@
 #     masks: Generator[MaskTile, None, None]
 #     metadata: Metadata
 
-#     def to_data_arrays(self) -> TiledDataArrayLayer:
-#         data_arrays = (mask.to_data_array(name=self.name) for mask in self.masks)
+#    def to_data_arrays(self) -> TiledDataArrayLayer:
+#        data_arrays = (mask.to_data_array(name=self.name) for mask in self.masks)
 
 #         return TiledDataArrayLayer(
 #             name=self.name,
 #             data_arrays=data_arrays,
 #             metadata=self.metadata,
 #         )
+
+
+# temporary func to read raster + resample & reshape if necessary
+from pathlib import Path
+from typing import List, Literal, Optional, Sequence, Union
+
+from rasterio.drivers import raster_driver_extensions
+from xarray import Dataset, open_dataset
+
+from sds_data_model._raster import _resample_and_reshape
+from sds_data_model._vector import _check_layer_projection
+
+
+def read_dataset_from_file(
+    data_path: str,
+    band: Union[List[int], List[str], None] = None,
+    categorical: Union[bool, Sequence[bool]] = False,
+    nodata: Optional[float] = None,
+    engine: Optional[str] = None,
+    decode_coords: Union[bool, None, Literal["coordinates", "all"]] = "all",
+) -> Dataset:
+    """# TODO.
+
+    Args:
+        data_path (str): #TODO
+        band (Union[List[int], List[str], None], optional): #TODO. Defaults to None.
+        categorical (Union[bool, Sequence[bool]], optional): #TODO. Defaults to False.
+        nodata (Optional[float], optional): #TODO. Defaults to None.
+        engine (Optional[str], optional): #TODO. Defaults to None.
+        decode_coords (Union[bool, None, Literal["coordinates", "all"]], optional):
+         #TODO. Defaults to "all".
+
+    Returns:
+        Dataset: #TODO
+    """
+    suffix = Path(data_path).suffixes[0]
+
+    if engine:
+        pass
+    elif suffix == ".zarr":
+        engine = "zarr"
+    elif suffix[1:] in raster_driver_extensions().keys():
+        engine = "rasterio"
+        decode_coords = None
+
+    dataset = open_dataset(
+        data_path,
+        engine=engine,
+        decode_coords=decode_coords,
+        mask_and_scale=False,
+    )
```

Comment:
There's nothing wrong with this by I prefer to distinguish between variables provided by the caller and variables created within the function, so I'd have gone for something like:

```suggestion
    if engine:
        _engine = engine
    elif suffix == ".zarr":
        _engine = "zarr"
        _decode_coords = decode_coords
    elif suffix[1:] in raster_driver_extensions().keys():
        _engine = "rasterio"
        _decode_coords = None
  
    dataset = open_dataset(
        data_path,
        engine=_engine,
        decode_coords=_decode_coords,
        mask_and_scale=False,
    )
```